tests/snapshot_tests.cc

```diff
@@ -0,0 +1,159 @@
+            expected.insert(de.name);
+            return make_ready_future<>();
+        }).get();
+        // snapshot triggered a flush and wrote the data down.
+        BOOST_REQUIRE_GT(expected.size(), 1);
+
+        // all files were copied and manifest was generated
+        lister::scan_dir((fs::path(cf.dir()) / "snapshots" / "test"), { directory_entry_type::regular }, [&expected] (fs::path parent_dir, directory_entry de) {
+            expected.erase(de.name);
+            return make_ready_future<>();
+        }).get();
+
+        BOOST_REQUIRE_EQUAL(expected.size(), 0);
+        return make_ready_future<>();
+    });
+}
+
+SEASTAR_TEST_CASE(snapshot_list_okay) {
+    return do_with_some_data([] (cql_test_env& e) {
+        auto& cf = e.local_db().find_column_family("ks", "cf");
+        take_snapshot(e).get();
+
+        auto details = cf.get_snapshot_details().get0();
+        BOOST_REQUIRE_EQUAL(details.size(), 1);
+
+        auto sd = details["test"];
+        BOOST_REQUIRE_EQUAL(sd.live, 0);
+        BOOST_REQUIRE_GT(sd.total, 0);
+
+        lister::scan_dir(fs::path(cf.dir()), { directory_entry_type::regular }, [] (fs::path parent_dir, directory_entry de) {
+            fs::remove(parent_dir / de.name);
+            return make_ready_future<>();
+        }).get();
+
+        auto sd_post_deletion = cf.get_snapshot_details().get0().at("test");
+
+        BOOST_REQUIRE_EQUAL(sd_post_deletion.total, sd_post_deletion.live);
+        BOOST_REQUIRE_EQUAL(sd.total, sd_post_deletion.live);
+
+        return make_ready_future<>();
+    });
+}
+
+SEASTAR_TEST_CASE(snapshot_list_inexistent) {
+    return do_with_some_data([] (cql_test_env& e) {
+        auto& cf = e.local_db().find_column_family("ks", "cf");
+        auto details = cf.get_snapshot_details().get0();
+        BOOST_REQUIRE_EQUAL(details.size(), 0);
+        return make_ready_future<>();
+    });
+}
+
+
+SEASTAR_TEST_CASE(clear_snapshot) {
+    return do_with_some_data([] (cql_test_env& e) {
+        take_snapshot(e).get();
+        auto& cf = e.local_db().find_column_family("ks", "cf");
+
+        unsigned count = 0;
+        lister::scan_dir((fs::path(cf.dir()) / "snapshots" / "test"), { directory_entry_type::regular }, [&count] (fs::path parent_dir, directory_entry de) {
+            count++;
+            return make_ready_future<>();
+        }).get();
+        BOOST_REQUIRE_GT(count, 1); // expect more than the manifest alone
+
+        e.local_db().clear_snapshot("test", {"ks"}).get();
+        count = 0;
+
+        BOOST_REQUIRE_EQUAL(fs::exists(fs::path(cf.dir()) / "snapshots" / "test"), false);
+        return make_ready_future<>();
+    });
+}
+
+SEASTAR_TEST_CASE(clear_nonexistent_snapshot) {
+    // no crashes, no exceptions
+    return do_with_some_data([] (cql_test_env& e) {
+        e.local_db().clear_snapshot("test", {"ks"}).get();
+        return make_ready_future<>();
+    });
+}
```

Comment:
Missing test for failed clear (directory contains other files).

Reply:
the directory containing other files do not cause the clear command to crash.
clear is akin to rm -rf, so it will succeed.

There are two situations that I can see in which it would crash:
1. when we obtained the list of files for deletion, we raced with another creation, and thus when time comes to delete the directory it is not empty and fail. (seen in the field).
2. some file inside that the user cannot delete (i.e., owned by root)

As I argued in the commit message, none of those are suitable for unit tests. The first is racy and will make our tests even flakier. 

The second would depend on external infrastructure, like having sudo access and being able to set xattr in the file to disable deletion (not available on tmpfs). Both would impose constraints on unit tests that I don't think are reasonable and are more suited for a DTest.

I wrote as much in the commit message